### PR TITLE
fix: add check if donation ticket has payment method enabled

### LINKED ISSUE #6788: user is able to save event without selecting payment method for donation ticket.
**Describe the bug**
Related Frontend issue - [#3850](https://github.com/fossasia/open-event-frontend/issues/3850)

Right now user is able to save event with `donation ticket` without selecting a `payment method`.

**Expected behavior**
Server should throw an error like `Donation ticket must have a payment method` as it throws in case of `Paid Tickets`

**Additional context**

I will work on this issue.

### PR DESCRIPTION
**Pull request by snitin315**
Fixes #6788 

#### Short description of what this resolves:

Server will throw an error `Donation ticket must have a payment method` as it throws in case of `Paid Tickets`

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- app/api/tickets.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4886967&bid=16188248)

**Comment by iamareebjamal** [comment]
Please run this and recommit:

```
pip install -r requirements/dev.txt
pre-commit install
black .
```

File: app/api/tickets.py [hunk 1]
```diff
@@ -248,6 +255,23 @@ def before_update_object(self, ticket, data, view_kwargs):
                     "Event having paid ticket must have a payment method",
                 )
 
+        if ticket.type == 'donation':
```
Comment (kushthedude) [inline app/api/tickets.py]: There is an operator called `or`. Remove all the duplicated code and merge with paid ticket

Comment (snitin315) [reply app/api/tickets.py]: What about Error message?

Comment (snitin315) [reply app/api/tickets.py]: should i change it to `Select a payment method`??

Comment (kushthedude) [reply app/api/tickets.py]: Donation is a paid ticket, Except the name and its function.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- app/api/tickets.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4886967&bid=16188248)

File: app/api/tickets.py (hunk 1 above)
Comment (kushthedude) [reply app/api/tickets.py]: > should i change it to `Select a payment method`??

Error message is fine, Also once you will make FE changes chances are less user will submit the form before completing the validations

Comment (snitin315) [reply app/api/tickets.py]: Done

File: app/api/tickets.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def before_create_object(self, data, view_kwargs):
                     {'event_id': data['event']}, "Event does not exist"
                 )
 
-            if data.get('type') == 'paid':
+            if data.get('type') == 'paid' or 'donation':
```
Comment (iamareebjamal) [inline app/api/tickets.py]: `data.get('type') == 'paid' or 'donation'`

`false or 'donation'`

`false or true`

`true`

Comment (snitin315) [reply app/api/tickets.py]: should I implement separate if block for `donation` and `paid` then?

Comment (iamareebjamal) [reply app/api/tickets.py]: No

**Pull request merged**